tablo/models.py

```diff
@@ -188,74 +213,190 @@ def time_info(self):
+                if token.ttype != Token.Name:
+                    where_clause += token.value
+                elif token.value != token.parent.value:
+                    # Token is aliased: just write the segments as they come
+                    where_clause += token.value.strip('"').join('""')
+                else:
+                    # Token is not aliased if parent doesn't include it: add source alias
+                    where_clause += '"source"."{field}"'.format(field=token.value.strip('"'))
+
+        query_params = []
+
+        if self.start_time_field:
+            layer_time_field = '"source"."{time_field}"'.format(time_field=self.start_time_field)
+
+            if start_time:
+                if start_time == end_time:
+                    where_clause += ' AND {time_field} = %s'.format(time_field=layer_time_field)
+                    query_params.append(start_time)
+                else:
+                    where_clause += ' AND {time_field} BETWEEN %s AND %s'.format(time_field=layer_time_field)
+                    query_params.append(start_time)
+                    query_params.append(end_time)
+            elif where is None and not count_only:
+                # If layer has a time component, default is to show first time step
+                where_clause += ' AND {time_field} = (SELECT MIN({subquery_time_field}) FROM "{table}")'.format(
+                    time_field=layer_time_field,
+                    subquery_time_field=self.start_time_field,
+                    table=self.table
+                )
+
+        if kwargs.get('object_ids'):
+            object_ids = kwargs['object_ids']
+
+            where_clause += ' AND "source"."{primary_key}" IN ({object_id_list})'.format(
                 primary_key=PRIMARY_KEY_NAME,
                 object_id_list=','.join(['%s' for obj_id in object_ids]),
             )
             for obj_id in object_ids:
-                params.append(obj_id)
+                query_params.append(obj_id)
 
-        if extent:
-            where += ' AND ST_Intersects(dbasin_geom, ST_GeomFromText(%s, 3857)) '
-            params.append(extent.as_sql_poly())
+        if kwargs.get('extent'):
+            where_clause += ' AND ST_Intersects("source"."dbasin_geom", ST_GeomFromText(%s, 3857)) '
+            query_params.append(kwargs['extent'].as_sql_poly())
 
-        with get_cursor() as c:
-            # where clause may contain references to table and fields
-            query_clause = 'SELECT {fields} FROM {table} WHERE ' + where
-            query_clause = query_clause.format(
-                fields=','.join(return_fields),
-                table=self.table
-            )
-            c.execute(query_clause, params)
-            response = dictfetchall(c)
+        return where_clause, query_params
 
-        return response
+    def _build_order_by_clause(self, fields, related_tables=None, limit=0, offset=1000):
```

Comment:
@MikeTheReader had the same concern. I addressed this as part of your previous comment about managing a default for limit. The new code just applies them to the query string before execution  in `perform_query` instead of depending on this or any other custom method. It seemed appropriate to handle them there because they don't really belong to a single common clause (like `select`, `join` or `order by`).